app/main.py

```diff
@@ -1,3 +1,53 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json") as file:
+        config = json.load(file)
+
+    fuel_price = config["FUEL_PRICE"]
+
+    customers = [
+        Customer(
+            config_customer["name"],
+            config_customer["product_cart"],
+            config_customer["location"],
+            config_customer["money"],
+            Car(**config_customer["car"])
+        )
+        for config_customer in config["customers"]
+    ]
+
+    shops = [Shop(**config_shop) for config_shop in config["shops"]]
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        trips = []
+
+        for shop in shops:
+            distance_to_shop = customer.distance_to(shop.location)
+            cost_to_shop = customer.travel_cost(distance_to_shop, fuel_price)
+            purchase_cost = customer.purchase_cost(shop.products)
+            total_trip_cost = 2 * cost_to_shop + purchase_cost
+            trips.append((total_trip_cost, shop))
+
+            print(f"{customer.name}'s trip to the {shop.name} "
+                  f"costs{total_trip_cost: .2f}")
+
+        trips.sort(key=lambda x: x[0])
+        cheapest_trip = trips[0]
+        total_cost, chosen_shop = cheapest_trip
+
+        if total_cost <= customer.money:
+            print(f"{customer.name} rides to {chosen_shop.name}\n")
+            customer.location = chosen_shop.location
+            customer.print_receipt(chosen_shop.name, chosen_shop.products)
+            customer.location = [0, 0]
```

Comment:
why is customer location equal to [0, 0], it should be equal to his initial location, he returns home, not to the point [0, 0]

Reply:
oh, my bad, i deleted it

Reply:
customer.location = chosen_shop.location
this line too